zjit/src/hir.rs

```diff
@@ -1717,8 +1723,35 @@ fn filter_translatable_calls(flag: u32) -> Result<(), ParseError> {
     Ok(())
 }
 
+#[derive(Debug)]
+struct ProfileOracle {
+    payload: &'static IseqPayload,
+    /// types is a map from insn_idx -> profiled type information at that idx
```

Comment:
```suggestion
    /// types is a map from ISEQ insn_idx -> profiled type information at that HIR idx
```

Is this understanding ^ right?

Both "insn_idx" and "that idx" are a kind of an "instruction index" (both of which _could_ be called `InsnId`), so I found this comment confusing, even after looking at the type.

Reply:
Well, I guess "that idx" may be pointing to "insn_idx", but then this comment doesn't explain what's `InsnId`. It'd be nice to clarify the difference between `usize` and `InsnId`.

Reply:
Ah, no, both refer to ISEQ -- thank you -- I should clarify

Reply:
At this point I want a type wrapper around iseq insn indices too but that is another PR

Reply:
sounds good. FYI it was called `IseqIdx` in YJIT.